### ISSUE TITLE
libp2p host key vs ACL

### ISSUE THREAD

**Issue by requilence** [textileio/go-threads#306]
In https://github.com/textileio/go-threads/pull/300 you've added the libp2p host public key to the first record in the log to match the log and the peer that own it. And according to the discussion in the https://github.com/textileio/go-threads/pull/293 it looks like you are going to implement thread ACL based on this node's pub key

So here is come the question. How are you going to handle __1 user__ – __multiple devices__ in that case? 

libp2p is not supposed to share identity between nodes: https://discuss.ipfs.io/t/can-multiple-nodes-share-the-same-peer-id/1204

Scenario:
- Thread owner grants the user permission to read and write to the thread
- Later this user want to access this thread from another client/device he/she owns

Questions:
- Who will be responsible for adding the other user device host ID to the ACL? 
- How to indicate that both logs/pubkeys belongs to the same user. Looks like it will be the must-have scenario for applications
- What if the user doesn't have ACL modify permission (e.g. can't add one more user with write permission)?

TLDR
- libp2p key, used in the 1st log's record looks like a device key, not a user key
- on the other hand looks like ACL should rely on the user itself, not on the device(libp2p host) key

At @anytypeio we've implemented the second layer cbornode on top of Record to store both device and user pubkey&signature. But it will be better to implement on the level of go-threads Logs. I think this is a pretty common use-case.

Maybe think to add the non-ephemeral user pubkey/signature alongside with the libp2p host key?

**Comment by sanderpick** [comment]
Hey @requilence, thanks for jumping in here! 

> Who will be responsible for adding the other user device host ID to the ACL?

> How to indicate that both logs/pubkeys belongs to the same user. Looks like it will be the must-have scenario for applications

> What if the user doesn't have ACL modify permission (e.g. can't add one more user with write permission)?

The `PubKey` in the first record is the public key associated with the user's `Identity`, which can be any king of key. Currently, the verifying end only knows how to handle libp2p type keys, but I'd like to extend this to whatever people need. For example, Keybase, ETH, 3Box, etc. So, it's not a device key really... it's an _identity_ that can be used across devices.

The ACL that we've been referring to is _fixed_. It's hash is part of the thread ID. So, changing ACL rules amounts to _forking_ the thread. The idea is that the ACL will contain public keys from the user identities.

**Comment by requilence** [comment]
@sanderpick thanks for the quick answer!
Maybe I got it wrong but here it seems like it sets from the ipfs-lite host peer https://github.com/textileio/go-threads/blob/master/net/net.go#L859

**Comment by sanderpick** [comment]
Oh, yes. In the case of `CreateRecord`, I assumed you're delegating your identity with the net host. `AddRecord` takes a pre-made record, in which you can specify the identity key: https://github.com/textileio/go-threads/blob/master/net/api/client/client_test.go#L255

However! There's no reason why the identity key can't be extracted from the thread token, which is optionally given to `CreateThread`. Sounds like a good idea.
